buildkite/resource_organization_settings.go

```diff
@@ -38,31 +38,27 @@ func resourceOrganizationSettings() *schema.Resource {
 	}
 }
 
-func assertError(err error) diag.Diagnostics {
-	if err != nil {
-		return diag.FromErr(err)
-	}
-	return nil
-}
-
 // CreateUpdateDeleteOrganizationSettings is used for the creation, updating and deleting of a Buildkite organization's settings
-// In the future, this will be split into separate functions, but given it only has one field, it's not worth it yet
 func CreateUpdateDeleteOrganizationSettings(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 	var diags diag.Diagnostics
 	client := m.(*Client)
 
 	response, err := getOrganization(client.genqlient, client.organization)
 
-	assertError(err)
+	if err != nil {
+		return diag.FromErr(err)
+	}
 
 	if response.Organization.Id == "" {
 		return diag.FromErr(fmt.Errorf("organization not found: '%s'", client.organization))
 	}
 
 	cidrs := strings.Join(d.Get("allowed_api_ip_addresses").([]string), " ")
-	apiResponse, queryError := setApiIpAddresses(client.genqlient, response.Organization.Id, cidrs)
+	apiResponse, err := setApiIpAddresses(client.genqlient, response.Organization.Id, cidrs)
 
-	assertError(queryError)
+	if err != nil {
+		return diag.FromErr(err)
+	}
```

Comment:
I don't really agree with the approach to this, given it causes repetition in the code. Consider refactoring if more errors need to be handled in the future.

Reply:
I think the problem with the function you had was that it would still allow continued execution which we don't want. If there's an error, we want to exit early, so you can't really get away without an `if` condition in the main function body